roles/2_setup-infra/tasks/subtasks/kernel.yml

```diff
@@ -0,0 +1,31 @@
+- name: Make specified kernel the default on boot
+  script: "files/set_default_kernel {{ sp_kernel_version }}"
+  when:
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+- name: Reboot host to apply specified kernel
+  reboot:
+    msg: Reboot initiated by Ansible
+    connect_timeout: 10
+    reboot_timeout: 600
+    pre_reboot_delay: 0
+    post_reboot_delay: 30
+    test_command: whoami
+  when:
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+# Not using ansible_kernel since the value does not update after reboot
+- name: Get current kernel
```

Comment:
It's better to make Ansible regather facts, the kernel command line might've changed.